guide/TEMPLATE.md

```diff
@@ -0,0 +1,53 @@
+# Using template from external repository
+
+## Repository structure
+
+The external repository must have a directory named ```template``` at the root directory, in which there are directories
+containing templates. The directory for each template can be freely named with alphanumeric characters and hyphen.
+
+Example:
+
+```
+template
+├── csharp
+│   ├── Dockerfile
+│   └── template.yml
+├── node
+│   ├── Dockerfile
+│   └── template.yml
+├── node-armhf
+│   ├── Dockerfile
+│   └── template.yml
+├── python
+│   ├── Dockerfile
+│   └── template.yml
+├── python-armhf
+│   ├── Dockerfile
+│   └── template.yml
+├── php-56
+│   ├── Dockerfile
+│   └── template.yml
+├── php-71
+│   ├── Dockerfile
+│   └── template.yml
+├── php-latest
+│   ├── Dockerfile
+│   └── template.yml
+└── ruby
+    ├── Dockerfile
+    └── template.yml
+```
+
+## Download external repository
+
+In order to build functions using 3rd party templates, you need to add 3rd templates before the build step, with the following command:
+
+```
+./faas-cli add-template https://github.com/alexellis/faas-cli
+```
+
+If you need to update the downloaded repository, just add the flag `--overwrite` to the download command:
+
+```
+./faas-cli add-template https://github.com/alexellis/faas-cli --override
+```
```

Comment:
Same idea here

Reply:
I guess it would be helpful if it was a different repo than the one we use for the project templates

Reply:
Incidentally this is a good way to update templates ^

Reply:
Yes - probably worth spelling that out in the guide too.

Reply:
I appreciate the input. Would this repo work?

https://github.com/ericstoekl/faas-custom-templates

Or would a copy of faas-cli work better? (https://github.com/ericstoekl/faas-cli)

Reply:
I'm for either https://github.com/ericstoekl/faas-custom-templates or https://github.com/itscaro/openfass-template-php, to show the way to add a community-supported languages.

Reply:
I think the former.  Really what we are trying to demonstrate is separation of repos so that its obvious to the user that they can point it at a repo that doesn't follow the structure of faas-cli. 

(also, the latter wouldn't work that well for ppl who have forked)

Reply:
Please update with the new syntax

Reply:
Let's use the template from @itscaro